### ISSUE TITLE
GetEnrollmentGroupAsync fails with attestation no contains root certificate

### ISSUE THREAD

**Issue by Banchio** [Azure/azure-iot-sdk-csharp#332]
I've created an enrollment group through the Azure Portal. When I try to use **GetEnrollmentGroupAsync** against the just created GroupId, I get the following exception: _ArgumentException: Attestation mechanism do not contains root certificate._
Note that the Rest API is invoked correctly, the error happens in the ctor of the enrollmentGroup:
```[JsonConstructor]
        internal EnrollmentGroup(
            string enrollmentGroupId,
            AttestationMechanism attestation,
            string iotHubHostName,
            TwinState initialTwinState,
            ProvisioningStatus provisioningStatus,
            DateTime createdDateTimeUtc,
            DateTime lastUpdatedDateTimeUtc,
            string eTag)
```

To add details, the exact exception is launched when setting the property Attestation of the group (below):
```
  [JsonIgnore]
        public Attestation Attestation
```

**Comment by mamokarz** [comment]
Hi @Banchio, thanks for use Microsoft SDK, I'll investigate this issue.

**Comment by mamokarz** [comment]
Hi @Banchio , this is a bug in the SDK, I'm fixing it.

**Comment by mamokarz** [comment]
Fixed by #339.
